From 68c477e02d727e5d894fe132602cf56841f9259c Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 16 May 2024 11:15:48 +0200
Subject: [PATCH 0866/1695] plat-stm32mp2: fix memory leak in
 scmi_scpfw_cfg_init()

When scmi_regulator_consumer_init(), free the allocated memory above.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 44b5181a7fa3c0687a76b1cfd74491e92fd89063 plat-stm32mp2: fix memory leak in scmi_scpfw_cfg_init()]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: Ie2f5808d9c3b9a21995128900b0864024f74d7f5
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/380121
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/scmi_server_scpfw.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/scmi_server_scpfw.c b/core/arch/arm/plat-stm32mp2/scmi_server_scpfw.c
index 6e5a0335a..104f3beb5 100644
--- a/core/arch/arm/plat-stm32mp2/scmi_server_scpfw.c
+++ b/core/arch/arm/plat-stm32mp2/scmi_server_scpfw.c
@@ -703,6 +703,8 @@ static TEE_Result scmi_scpfw_cfg_init(void)
 
 		res = scmi_regulator_consumer_init();
 		if (res) {
+			free(channel_cfg->reset);
+			free(channel_cfg->clock);
 			EMSG("SCMI regulator consumer init: %#"PRIx32, res);
 			return res;
 		}
-- 
2.47.2

